baybe/objectives/single.py

```diff
@@ -70,6 +73,26 @@ def to_botorch(self) -> MCAcquisitionObjective:
 
         return IdentityMCObjective()
 
+    @override
+    def to_botorch_posterior_transform(self) -> ScalarizedPosteriorTransform | None:
+        if not (
+            isinstance((t := self._target), NumericalTarget)
+            and isinstance(
+                (tr := t.transformation), (IdentityTransformation, AffineTransformation)
+            )
+        ):
+            raise NonGaussianityError(
+                f"Converting an objective of type '{type(self).__name__}' is only "
+                f"possible when the transformation result is Gaussian, that is, "
+                f"when the target is of type '{NumericalTarget.__name__}' and the "
+                f"assigned transformation is affine."
+            )
+
+        if isinstance(tr, IdentityTransformation) and not t.minimize:
```

Comment:
@Scienfitz: I've added a commit that implements the shortcut of representing "no transformation" (a.k.a identity) as `None`, since one can argue it saves one layer of function calls in this special case. However, as you can see in the last commit, I immediately fell into the trap and forgot to add the `and` part, which I fixed afterwards. This makes me question if it's really worth doing the special handling. It also has the second disadvantage that we get a union return type, meaning that the caller always has do add the extra `if None` logic. So right now, I'm tempted to remove it again, because this case basically shows that it's more error prone, but wanted to hear your thoughts.

Reply:
removed